csr_check.go

```diff
@@ -119,7 +124,14 @@ func validateCSRContents(req *certificatesv1beta1.CertificateSigningRequest, csr
 //
 // For server certificates:
 // Names contained in the CSR are checked against addresses in the corresponding node's machine status.
-func authorizeCSR(config ClusterMachineApproverConfig, machines []v1beta1.Machine, nodes corev1client.NodeInterface, req *certificatesv1beta1.CertificateSigningRequest, csr *x509.CertificateRequest) error {
+func authorizeCSR(
+	config ClusterMachineApproverConfig,
+	machines []v1beta1.Machine,
+	nodes corev1client.NodeInterface,
+	req *certificatesv1beta1.CertificateSigningRequest,
+	csr *x509.CertificateRequest,
+	ca *x509.CertPool,
+) error {
 	if len(machines) == 0 || req == nil || csr == nil {
 		return fmt.Errorf("Invalid request")
```

Comment:
This check `len(machines) == 0 ` would defeat the intention of the PR.

Reply:
Good call. I hadn't actually run this outside AWS. Removed that check.